src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers/ApiControllerAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+
+        private static readonly LocalizableString Title = new LocalizableResourceString(nameof(Resources.ApiControllerDescription), Resources.ResourceManager, typeof(Resources));
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(Resources.ApiControllerDescription), Resources.ResourceManager, typeof(Resources));
+        private static readonly LocalizableString Description = new LocalizableResourceString(nameof(Resources.ApiControllerDescription), Resources.ResourceManager, typeof(Resources));
+
+        private static readonly DiagnosticDescriptor Rule = new(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterSyntaxNodeAction(AnalyzeNode, cs.SyntaxKind.SimpleBaseType);
+            context.RegisterSyntaxNodeAction(AnalyzeNode, vb.SyntaxKind.InheritsStatement);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var namedTypeSymbol = context.ContainingSymbol as INamedTypeSymbol;
+
+            if (namedTypeSymbol != null && namedTypeSymbol.BaseType != null
+                && namedTypeSymbol.BaseType.ToDisplayString().Equals($"{BadNamespace}.{BadClassName}", StringComparison.Ordinal))
+            {
+                if (context.Compilation.Language == LanguageNames.CSharp)
+                {
+                    ReportCSharpSyntax(context);
+                }
+                else if (context.Compilation.Language == LanguageNames.VisualBasic)
+                {
+                    ReportVisualBasicSyntax(context);
+                }
+            }
+        }
+
+        private static void ReportCSharpSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var diagnostic = Diagnostic.Create(Rule, context.Node.GetLocation(), context.Node.ToString());
+            context.ReportDiagnostic(diagnostic);
+        }
+
+        /// <summary>
+        /// Uses syntax analysis to highlight the correct part of the node. At this point
+        /// the context.Node looks like 'Inherits System.Web.Http.ApiController' and we
+        /// do not want to highlight the VisualBasic.SyntaxKind.InheritsKeyword
+        /// </summary>
+        /// <param name="context"></param>
+        private static void ReportVisualBasicSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var baseClass = context.Node.DescendantNodes()
```

Comment:
Language agnostic means that you can't reference any of the c# or vb roslyn assemblies. That can cause problems in cases where the compiler does not load the vb stuff or the cs stuff.